lib/components/soundIcon/soundIcon.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:redux/redux.dart';
+import '../../appState.dart';
+import '../../utils/challengeUtil.dart';
+import '../../actions/actions.dart';
+
+class SoundIcon extends StatelessWidget {
+  final Set<String> achivements = Set.from(['1.2', '1.1']);
+
+  @override
+  Widget build(BuildContext context) {
+    return StoreConnector(
+      converter: _ViewModel.fromStore,
+      builder: (context, vm) {
+        String latestAchivement =
+            getLatestAchivement(vm.completeAchivements, achivements);
+        switch (latestAchivement) {
+          case '1.1':
+            return _StaticMutedSoundIcon();
+          case '1.2':
+            return _ClickableSoundIcon();
+          default:
+            return Column();
+        }
+      },
+    );
+  }
+}
+
+class _StaticMutedSoundIcon extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+      icon: Icon(Icons.volume_off),
+      iconSize: 40,
+      padding: EdgeInsets.all(24),
+      onPressed: () => null,
+    );
+  }
+}
+
+class _ClickableSoundIcon extends StatefulWidget {
+  @override
+  _ClickableSoundIconState createState() => _ClickableSoundIconState();
+}
+
+class _ClickableSoundIconState extends State<_ClickableSoundIcon> {
```

Comment:
Does this need to be a StatefulWidget? 🤔

Reply:
nope :)